src/components/RequesterPaysModal.js

```diff
@@ -0,0 +1,70 @@
+import { div, h } from 'react-hyperscript-helpers'
+import { buttonPrimary, link, spinnerOverlay } from 'src/components/common'
+import { icon } from 'src/components/icons'
+import Modal from 'src/components/Modal'
+import * as Nav from 'src/libs/nav'
+import { authStore, freeCreditsActive } from 'src/libs/state'
+import { Component } from 'src/libs/wrapped-components'
+import * as Utils from 'src/libs/utils'
+
+
+export default Utils.connectAtom(authStore, 'authState')(class RequesterPaysModal extends Component {
+  render() {
+    const { authState: { profile } } = this.props
+    const { trialState } = profile
+    const hasFreeCredits = trialState === 'Enabled'
+    const loading = false
+    const onDismiss = this.props
+
+    return Utils.cond(
+      [loading, spinnerOverlay],
+      [hasFreeCredits, () => h(Modal, {
+        title: 'Set up Billing',
+        onDismiss,
+        showCancel: false,
```

Comment:
We may want to consider adding a cancel button that executes onDismiss, this will be consistent with the other RP modals